cmd/tnf/check/results/results.go

```diff
@@ -0,0 +1,216 @@
+	if generateTemplate {
+		return generateTemplateFile(actualTestResults)
+	}
+
+	// Get the expected test results from the reference YAML template
+	expectedTestResults, err := getExpectedTestResults(templateFileName)
+	if err != nil {
+		return fmt.Errorf("could not get the expected test results, err: %v", err)
+	}
+
+	// Match the results between the test results DB and the reference YAML template
+	var mismatchedTestCases []string
+	for testCase, testResult := range actualTestResults {
+		if testResult != expectedTestResults[testCase] {
+			mismatchedTestCases = append(mismatchedTestCases, testCase)
+		}
+	}
+
+	// Verify that there are no unmatched expected test results
+	for testCase := range expectedTestResults {
+		if _, exists := actualTestResults[testCase]; !exists {
+			mismatchedTestCases = append(mismatchedTestCases, testCase)
+		}
+	}
+
+	if len(mismatchedTestCases) > 0 {
+		fmt.Println("Expected results DO NOT match actual results")
+		printTestResultsMismatch(mismatchedTestCases, actualTestResults, expectedTestResults)
+		os.Exit(1)
+	}
+
+	fmt.Println("Expected results and actual results match")
+
+	return nil
+}
+
+func getTestResultsDB(logFileName string) (map[string]string, error) {
+	resultsDB := make(map[string]string)
+
+	file, err := os.Open(logFileName)
+	if err != nil {
+		return nil, fmt.Errorf("could not open file %q, err: %v", logFileName, err)
+	}
+	defer file.Close()
+
+	re := regexp.MustCompile(`.*\[(.*?)\]\s+Recording result\s+"(.*?)"`)
+
+	scanner := bufio.NewScanner(file)
+	for scanner.Scan() {
+		line := scanner.Text()
+		match := re.FindStringSubmatch(line)
+		if match != nil {
+			testCaseName := match[1]
+			result := match[2]
+			resultsDB[testCaseName] = result
+		}
+	}
+
+	if err := scanner.Err(); err != nil {
+		return nil, fmt.Errorf("error scanning file, err: %v", err)
+	}
+
+	return resultsDB, nil
+}
+
+func getExpectedTestResults(templateFileName string) (map[string]string, error) {
+	templateFile, err := os.ReadFile(templateFileName)
+	if err != nil {
+		return nil, fmt.Errorf("could not open template file %q, err: %v", templateFileName, err)
+	}
+
+	var expectedTestResultsList TestResults
+	err = yaml.Unmarshal(templateFile, &expectedTestResultsList)
+	if err != nil {
+		return nil, fmt.Errorf("could not parse the template YAML file, err: %v", err)
+	}
+
+	expectedTestResults := make(map[string]string)
+	for _, testCase := range expectedTestResultsList.Pass {
+		expectedTestResults[testCase] = resultPass
+	}
+	for _, testCase := range expectedTestResultsList.Skip {
+		expectedTestResults[testCase] = resultSkip
+	}
+	for _, testCase := range expectedTestResultsList.Fail {
+		expectedTestResults[testCase] = resultFail
+	}
+
+	return expectedTestResults, nil
+}
+
+func printTestResultsMismatch(mismatchedTestCases []string, actualResults, expectedResults map[string]string) {
+	fmt.Printf("\n")
+	fmt.Println(strings.Repeat("-", 96)) //nolint:gomnd // table line
```

Comment:
optional: there are libraries automating text tables in go. For instance, the one used in tasty: https://github.com/syohex/go-texttable

Reply:
Thanks for the suggestion. But I think it's better not to have this dependency for the limited use of tables we have in our project.